app/src/main/java/com/example/weather/ui/main/MainActivity.java

```diff
@@ -39,10 +49,47 @@ private void settingsToolbar() {
         getSupportActionBar().setDisplayShowHomeEnabled(true);
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.main_menu,menu);
+        return true;
+    }
 
     @Override
-    public boolean onSupportNavigateUp() {
-        onBackPressed();
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.russian:
+                setLocale("ru");
+                recreate();
+                break;
+            case R.id.english:
+                setLocale("en");
+                recreate();
+                break;
+            case android.R.id.home:
+                onBackPressed();
+                break;
+
+        }
         return true;
     }
+
+    void setLocale(String lang){
```

Comment:
This methods should be inside separate class like "SettingManager" or "LocaleManager"